--- fitz.py	2023-11-30 19:21:20
+++ fitz1.py	2023-11-28 22:38:27
@@ -1861,6 +1861,14 @@
         if APN[0] == "dict":
             nstates = []
             APN = APN[1][2:-2]
+            apnt = APN.split("/")[1:]
+            for x in apnt:
+                nstates.append(x.split()[0])
+            states["normal"] = nstates
+        if APN[0] == "xref":
+            nstates = []
+            nxref = int(APN[1].split(" ")[0])
+            APN = doc.xref_object(nxref)
             apnt = APN.split("/")[1:]
             for x in apnt:
                 nstates.append(x.split()[0])
@@ -1869,6 +1877,14 @@
         if APD[0] == "dict":
             dstates = []
             APD = APD[1][2:-2]
+            apdt = APD.split("/")[1:]
+            for x in apdt:
+                dstates.append(x.split()[0])
+            states["down"] = dstates
+        if APD[0] == "xref":
+            dstates = []
+            dxref = int(APD[1].split(" ")[0])
+            APD = doc.xref_object(dxref)
             apdt = APD.split("/")[1:]
             for x in apdt:
                 dstates.append(x.split()[0])
